src/utils.ts

```diff
@@ -53,32 +64,40 @@ export const extractSecret = (
 	core.setSecret(secretValue);
 };
 
-export const loadSecrets = async (shouldExportEnv: boolean): Promise<void> => {
-	// Pass User-Agent Information to the 1Password CLI
-	setClientInfo({
-		name: "1Password GitHub Action",
-		id: "GHA",
-		build: semverToInt(version),
-	});
-
-	// Load secrets from environment variables using 1Password CLI.
-	// Iterate over them to find 1Password references, extract the secret values,
-	// and make them available in the next steps either as step outputs or as environment variables.
-	const res = await exec.getExecOutput(`sh -c "op env ls"`);
+export const loadSecrets = async (
+	auth: SecretReferenceResolver,
+	shouldExportEnv: boolean,
+): Promise<void> => {
+	const refs = loadSecretRefsFromEnv();
 
-	if (res.stdout === "") {
+	if (refs.length === 0) {
 		return;
 	}
 
-	const envs = res.stdout.replace(/\n+$/g, "").split(/\r?\n/);
-	for (const envName of envs) {
-		extractSecret(envName, shouldExportEnv);
+	for (const key of refs) {
+		await extractSecret(auth, key, shouldExportEnv);
```

Comment:
Can we use `await Promise.all()` here?

Reply:
Hi, I'm guessing it's not a blocker, right? This PR is quite large now, how about move all non-blocker changes to separate PRs instead?

Reply:
Cool